packages/flutter/lib/src/material/material.dart

```diff
@@ -80,23 +80,19 @@ const Map<MaterialType, BorderRadius?> kMaterialEdges = <MaterialType, BorderRad
 /// An interface for creating [InkSplash]s and [InkHighlight]s on a [Material].
 ///
 /// Typically obtained via [Material.of].
-abstract class MaterialInkController {
-  /// The color of the material.
-  Color? get color;
-
-  /// The ticker provider used by the controller.
-  ///
-  /// Ink features that are added to this controller with [addInkFeature] should
-  /// use this vsync to drive their animations.
-  TickerProvider get vsync;
+//
+// TODO(nate-thegrate): deprecate this in the next PR!
+// https://github.com/flutter/flutter/pull/149963
+typedef MaterialInkController = SplashController;
 
+/// Allows a [MaterialInkController] to call the [addInkFeature] method.
+///
+/// Will be deprecated once the [Splash] class is implemented.
+extension AddInkFeature on MaterialInkController {
```

Comment:
From the [style guide](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-extension):

> ### Avoid using `extension`.
> Extension methods are confusing to document and discover. To an end developer, they appear no different than the built-in API of the class, and discovering the documentation and implementation of an extension is more challenging than for class members.
>
> Prefer instead adding methods directly to relevant classes. If that is not possible, create a method that clearly identifies what object(s) it works with and is part of.
>
> (A rare exception can be made for extensions that provide temporary workarounds when deprecating features. In those cases, however, the extensions and all their members must be deprecated in the PR that adds them, and they must be removed in accordance with our deprecation policy.)

In my mind, there are a few options:

1. Don't add this extension: instead, introduce SplashController with a deprecated `addInkFeature` method that points to `addSplash`
2. Don't change `addInkFeature` in this PR (wait to add the extension until the actual migration)
3. Add the extension now, so that SplashController can simply be picked up and moved to the widgets layer.

At this point I'm going with number 3 but am open to suggestions!

Reply:
With the current approach (3), there's some risk that a user will start depending on addInkFeature without seeing that it will be deprecated. Options 1 and 2 don't have this problem.

Actually can you explain the final state of addSplash and SplashController after all PRs are landed and deprecations removed? To make sure I'm seeing the big picture.

Reply:
Currently, we have an **_InkFeatures** widget that manages the **_RenderInkFeatures** render object.

**Material** is structured as a stateful widget, since it stores a global key for **_InkFeatures** and acts as a ticker provider for it as well.

Eventually, the goal is to have a **SplashBox** widget that acts as a minimal wrapper for **_InkFeatures** (or whatever we rename it to), and **Material** will be a stateless widget that uses **SplashBox** in its build method.

<br>

As far as **_RenderInkFeatures** is concerned, here's the plan:
- Change name to **SplashController**
- Remove the useless `absorbHitTest` flag
- Rename `addInkFeature` to `addSplash`

Functionally, there's no difference between keeping `addInkFeature` in the **SplashController** class and deprecating it, vs. converting to an extension method. But it might be nice for all the material-specific names to stay in the material library.

<br>

> there's some risk that a user will start depending on `addInkFeature` without seeing that it will be deprecated

I agree that there's some risk, but the only reason a developer would manually call `addInkFeature` is if they want to ignore `InkWell` entirely and implement their own "touch response" widget from scratch.
(If we'd like to deprecate `addInkFeature` in this PR, that would work too.)

Reply:
Oh, I forgot to mention: currently, **MaterialInkController** is a public interface that is implemented by **_RenderInkFeatures**.

IMO the "public interface/private implementation" paradigm can be great in a lot of situations—in the case of the ink controller, I can see the appeal of not having all those render object methods in your autofill suggestions (the only alternative is to add `@protected` to all the methods you want to hide).
But since the interface is essentially not used at all, I feel that the only noticeable impact of this structure is how ugly it makes all those tests.